src/asm_state_machine.h

```diff
@@ -9,24 +9,43 @@
 #pragma once
 
 #include "slots_tracker.h"
+#include "util/khash.h"
+#include "deps/rmutil/rm_assert.h"
+#include <stdatomic.h>
+#include <stdlib.h>
+#include <pthread.h>
 
-#ifdef __cplusplus
-extern "C" {
+// Sanitizer detection for leak tracking
+#define ASM_SANITIZER_ENABLED 0
+#if defined(__has_feature)
+# if __has_feature(address_sanitizer)
+#  define ASM_SANITIZER_ENABLED 1
+# endif
+#elif defined(__SANITIZE_ADDRESS__)
+# define ASM_SANITIZER_ENABLED 1
 #endif
 
+#if ASM_SANITIZER_ENABLED
+// Dynamic array to track allocated pointers for leak detection
```

Comment:
Can you explain why we need that for sanitizer to detect leaks properly? Is it related to the fact that we are using globals here?

Reply:
So the idea is, by taking profit from the SANITIZER mechanism, we may be able to detect failures where we miss some counting and discounting of queries. This is a way to be able to validate this functionality in every test and PR, instead of only knowing from test ASM